design_documents/transactional_testing/transactional_testing.md

```diff
@@ -0,0 +1,75 @@
+# Transactional testing proposal
+
+## Context
+
+Some flows are not currently intuitively testable via standard testing methods, 
+like accounts logic (__validate__, __execute__ etc.), transaction rejections,
+fees, contract upgrade flows, and other e2e testing procedures.
+
+## Goal
+
+In order to facilitate that, we should consider adding a way to test
+the contracts in conditions as close to real networks conditions as possible.
+
+Also, the goal is to make the deployments testable and validation 
+of the deployments possible to the user. 
+
+The tests should be also runnable in a variety of environments, depending on the
+use case (namely katana, devnet, or testnet). This would increase user's confidence
+in the release process.
+
+## Proposed solution
+
+We could consider extending `cast script`, since the functionalities would overlap.
+This could be considered as a variant of `cast script`, in a way. 
+
+The extending/differentiating factors would be:
+
+- Extra environment-specific functions (for katana/devnet)
+- Additional RPC functions support (receipts, etc.)
+- Test-like behavior (fail/pass)
+
+
+## Solution architecture
+
+This is pretty straightforward. We ought to use the existing architecture for hint interception,
+provide extra functionality via libraries (one for forge, second one for devnet), and collect the cases from
```

Comment:
>one for forge, second one for devnet

Wdym by that?

Reply:
Ah brain fart, i meant katana and devnet